### PR TITLE
Kyle's RPS

### PR DESCRIPTION
**Pull request by KyleMacPherson**

### CODE REVIEW

**Comment by KyleMacPherson** [comment]
Brandon recommended bootstrap and advised to be wary of multiple elusive statements

File: capybara-201509191709327671017857.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div><center><font size=10>
+  Enter your move! Rock, paper or scissors?!</font><center></div>
+
+  <div>
```
Comment (pitchinvasion) [inline capybara-201509191709327671017857.html]: Delete these!

File: lib/computer.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Computer
+
+  attr_reader :moves_array
+
+  def initialize
+    @moves_array = [:rock, :paper, :scissors]
```
Comment (pitchinvasion) [inline lib/computer.rb]: If this never changes, you could extract this into a constant, rather than using a variable

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+class Game
+
+  attr_reader :rock_hash, :paper_hash, :scissors_hash
+
+def initialize
+  @rock_hash = {:paper => "PAPER WINS", :scissors => "ROCK WINS", :rock => "DRAW", :lizard => "ROCK WINS", :spock => "SPOCK WINS"}
```
Comment (pitchinvasion) [inline lib/game.rb]: Same as above, extract to constant

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  @paper_hash = {:paper => "DRAW", :scissors => "SCISSORS WIN", :rock => "PAPER WINS", :lizard => "LIZARD WINS", :spock => "PAPER WINS"}
+  @scissors_hash = {:paper => "SCISSORS WIN", :scissors => "DRAW", :rock => "ROCK WINS", :lizard => "SCISSORS WIN", :spock => "SPOCK WINS" }
+  @spock_hash = {:paper => "PAPER WINS", :scissors => "SPOCK WINS", :rock => "SPOCK WINS", :lizard => "LIZARD WINS", :spock => "DRAW"}
+  @lizard_hash = {:paper => "LIZARD WINS", :scissors => "SCISSORS WIN", :rock => "ROCK WINS", :lizard => "DRAW", :spock => "LIZARD WINS"}
+end
+
+def result player1_move, player2_move
+  if player1_move == 'rock'
+    result = @rock_hash[player2_move.to_sym]
```
Comment (pitchinvasion) [inline lib/game.rb]: Could you refactor this so it didn't use lots of if/else? This sort of thing would be frowned upon in a tech test (as are case statements), as there is almost always a simpler solution - and there's definitely ways of doing this which would avoid this and mean you don't have all those different instance variables

File: lib/player.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# class Player
+#
+#   attr_reader :possible_moves, :move, :name
+#
```
Comment (pitchinvasion) [inline lib/player.rb]: Delete this

File: lib/rps_web.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  get '/result' do
+    names = %w(rock paper scissors lizard spock)
+    unless names.include? params[:move]
+      session[:error] = "#{params[:move]} is not a valid move try again"
+      redirect '/form'
+    end
+    @name = params[:name]
+    @move = params[:move].downcase
+    game = Game.new
```
Comment (pitchinvasion) [inline lib/rps_web.rb]: There's too much logic here in your controller, is there a way you can extract this out into your libraries? The controller's responsibility should be to do as little as possible

File: lib/rps_web.rb [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    get '/2pform' do
+      @error = session[:error]
+      erb :twoplayerform
+    end
+
+    post '/submit_name' do #this is telling it what to do when it receives a post request to /submit_name
+      names = %w(rock paper scissors lizard spock)
+      unless names.include? params[:move]
+        session[:error] = "#{params[:move]} is not a valid move try again"
+        redirect '/2pform'
```
Comment (pitchinvasion) [inline lib/rps_web.rb]: Again this is is very messy, if you were coming to this code that someone else had written, would you consider this clear? How would you make it clearer?

File: spec/computer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+require 'computer.rb'
+
+describe Computer do
+
+  #it 'chooses rock, paper or scissors' do
+    #expect(subject.play).to
```
Comment (pitchinvasion) [inline spec/computer_spec.rb]: No tests? You need to do TDD! It's basically the main thing we're looking for

File: spec/game_spec.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+describe Game do
+
+ it 'returns the winning move based on both players moves' do
+   game = Game.new
+   expect(game.result('rock', 'scissors')).to eq('ROCK WINS')
+ end
+
+ it 'returns the winning move based on both players moves including lizard/spock' do
+   game = Game.new
```
Comment (pitchinvasion) [inline spec/game_spec.rb]: how is this different from the first test? are you checking both players here?

File: views/index.erb [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<head>
+<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
+</head>
+<body>
+<div>
+
+  <font size = 10> <center> Hello! Welcome to Rock Paper Scissors Lizard Spock.
```
Comment (pitchinvasion) [inline views/index.erb]: These tags are old and deprecated, you should use semantic HTML and use CSS for styling

**Comment by pitchinvasion** [comment]
It's good to see you've got this working and got all your feature tests written and passing, but this needs some work in terms of the game logic - by putting more logic out of your controllers and into your other classes you'll see the code clean up, this will also allow you to write more unit tests to test all your logic and will mean you're really taking advantage of OOP and things like single responsibility principle which is crucial for you to understand on your way to becoming an awesome junior developer. :construction:

**Pull request closed without merging**